conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptionHandler.java

```diff
@@ -16,37 +16,14 @@
 
 package com.palantir.conjure.java.undertow.runtime;
 
-import com.palantir.conjure.java.api.errors.SerializableError;
-import com.palantir.conjure.java.undertow.lib.Serializer;
-import io.undertow.server.HttpHandler;
+import com.palantir.conjure.java.undertow.lib.ExceptionHandler;
 import io.undertow.server.HttpServerExchange;
 
-/**
- * Delegates to the given {@link HttpHandler}, and catches&forwards all {@link Throwable}s. Any exception thrown in the
- * delegate handler is caught and serialized using the Conjure JSON format into a {@link SerializableError}. The result
- * is written it into the exchange's output stream, and an appropriate HTTP status code is set.
- */
-final class ConjureExceptionHandler implements HttpHandler {
-
-    private final Serializer<SerializableError> serializer;
-    private final HttpHandler delegate;
-
-    ConjureExceptionHandler(HttpHandler delegate) {
-        this(delegate, ConjureExceptions.serializer());
-    }
-
-    // Constructor allows new exception handlers to be created without creating new serializer instances.
-    ConjureExceptionHandler(HttpHandler delegate, Serializer<SerializableError> serializer) {
-        this.delegate = delegate;
-        this.serializer = serializer;
-    }
+enum ConjureExceptionHandler implements ExceptionHandler {
+    INSTANCE;
 
     @Override
-    public void handleRequest(HttpServerExchange exchange) {
-        try {
-            delegate.handleRequest(exchange);
-        } catch (Throwable throwable) {
-            ConjureExceptions.handle(exchange, serializer, throwable);
-        }
+    public void handle(HttpServerExchange exchange, Throwable throwable) {
+        ConjureExceptions.handle(exchange, throwable);
```

Comment:
Any reason to keep the exception handling logic in a separate class? To augment the logic we should be able to define our own `ExceptionHandler` that takes this a delegate

Reply:
Yep, I've updated ConjureExceptions to an enum singleton which implements ExceptionHandler in order to avoid changing our most heavily used logger origin.